File: python_backend/app/application/use_cases/get_document.py
"""Fetch a single document by id."""
from app.application.ports.documents_repository import DocumentsRepository
from app.domain.entities.document import Document
from app.domain.exceptions import NotFoundError


class GetDocumentUseCase:
    def __init__(self, repo: DocumentsRepository) -> None:
        self.repo = repo

    def execute(self, *, user_id: str, doc_id: int) -> Document:
        doc = self.repo.get(doc_id, user_id)
        if doc is None:
            raise NotFoundError("no encontrado")
        return doc

File: apps/web/src/application/use-cases/get-document.ts
/**
 * GetDocumentUseCase — fetch a single document with full content (FR-36).
 *
 * Returns the domain `Document` shape (camelCase). 404s surface as Error
 * from the adapter; the caller decides UI treatment.
 */

import type { DocumentsApiPort } from "@/application/ports/documents-api.port";
import type { Document } from "@/domain/entities/document";

export class GetDocumentUseCase {
  constructor(private readonly api: DocumentsApiPort) {}

  execute(id: number): Promise<Document> {
    return this.api.get(id);
  }
}
